src/hddtemp.c

```diff
@@ -158,16 +166,41 @@ static int hddtemp_query_daemon (char *buffer, int buffer_size)
 	if (fd < 0)
 	{
 		ERROR ("hddtemp plugin: Could not connect to daemon.");
-		return (-1);
+		return (NULL);
 	}
 
 	/* receive data from the hddtemp daemon */
-	memset (buffer, '\0', buffer_size);
-
+	buffer = NULL;
+	buffer_size = 0;
 	buffer_fill = 0;
-	while ((status = read (fd, buffer + buffer_fill, buffer_size - buffer_fill)) != 0)
+	while (1)
 	{
-		if (status == -1)
+		if (buffer_fill >= buffer_size - 1)
```

Comment:
This will stop working if somebody decides to use size_t for buffer_fill and buffer_size (which sounds like a good choice to me in general). Then, if buffer_size == 0, buffer_size -1 will underflow. To fix this, either handle this case separately here or pre-allocate 1024 bytes before entering the loop and then drop the buffer_size == 0 case below.